File: backend/usecase/getUser.go
package usecase

import (
	"context"
	"fmt"
	"net/http"

	"github.com/sijysn/resistar/backend/graph/model"
	"github.com/sijysn/resistar/backend/internal/auth"
	"github.com/sijysn/resistar/backend/internal/middleware"
	"github.com/sijysn/resistar/backend/repository"
	"github.com/sijysn/resistar/backend/utility"
)

func(u *UsecaseRepository) GetUser(ctx context.Context, input model.UserQuery) (*model.User, error) {
	responseAccess := ctx.Value(middleware.ResponseAccessKey).(*middleware.ResponseAccess)
	if responseAccess.Status == http.StatusInternalServerError {
		return nil, fmt.Errorf("サーバーエラーが発生しました")
	}
	if responseAccess.Status != auth.StatusGroup {
		errorMessage := "認証されていません"
		return &model.User {
			ErrorMessage: &errorMessage,
		}, nil
	}
	userID, err := utility.ParseStringToUint(input.UserID)
	if err != nil {
		return nil, err
	}
	getUserInput := repository.GetUserInput{
		UserID: userID,
	}
	dbUser, err := u.Repository.GetUser(getUserInput)
	user := &model.User{
		ID:       utility.ParseUintToString(dbUser.ID),
		Email:    dbUser.Email,
		Name:     dbUser.Name,
		ImageURL: dbUser.ImageURL,
	}
	return user, nil
}

File: frontend/lib/apollo/api/getUser.ts
import { gql } from "@apollo/client";

export const GET_USER = gql`
  query GetUser($userID: ID!) {
    user(input: { userID: $userID }) {
      id
      name
      email
      imageURL
    }
  }
`;

export type UserProps = {
  id: string;
  name: string;
  email: string;
  imageURL?: string;
};

export type getUserProps = {
  user: UserProps;
};

export type getUserVarsProps = {
  userID: string;
};
